internal/liqonet/tunnel-operator/offloaded_pod_controller.go

```diff
@@ -0,0 +1,171 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tunneloperator
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	liqoconsts "github.com/liqotech/liqo/pkg/consts"
+	liqoipset "github.com/liqotech/liqo/pkg/liqonet/ipset"
+	liqoiptables "github.com/liqotech/liqo/pkg/liqonet/iptables"
+	liqovk "github.com/liqotech/liqo/pkg/virtualKubelet/forge"
+)
+
+// OffloadedPodController reconciles an offloaded Pod object
+type OffloadedPodController struct {
+	client.Client
+	liqoiptables.IPTHandler
+	*liqoipset.IPSetHandler
+
+	// Liqo Gateway network namespace
+	gatewayNetns ns.NetNS
+
+	// Local cache of podInfo objects
+	podsInfo *sync.Map
+
+	// Local cache of endpointsliceInfo objects
+	endpointslicesInfo *sync.Map
+}
+
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;
+//+kubebuilder:rbac:groups=core,resources=pods/status,verbs=get;
+
+// NewOffloadedPodController instantiates and initializes the offloaded pod controller.
+func NewOffloadedPodController(cl client.Client, gatewayNetns ns.NetNS, podsInfo *sync.Map, endpointslicesInfo *sync.Map) (*OffloadedPodController, error) {
+	// Create the IPTables handler
+	iptablesHandler, err := liqoiptables.NewIPTHandler()
+	if err != nil {
+		return nil, err
+	}
+	// Create the IPSet handler
+	ipsetHandler := liqoipset.NewIPSetHandler()
+	// Create and return the controller
+	return &OffloadedPodController{
+		Client:             cl,
+		IPTHandler:         iptablesHandler,
+		IPSetHandler:       &ipsetHandler,
+		gatewayNetns:       gatewayNetns,
+		podsInfo:           podsInfo,
+		endpointslicesInfo: endpointslicesInfo,
+	}, nil
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Pod object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
```

Comment:
I think this comment don't need to be so verbose. It should be more specific for Liqo, describing what the controller purpose is.